### PR TITLE
Refining automated documentation Part 1/n

### LINKED ISSUE #595: WaterTAP Documentation for Zero-order Models
Checklist for refining zero-order model documentation. Technically, we have models via WT3 and AMO, though our initial focus is on finalizing WT3 derived models.
**High Priority:**
- [x] Edit automated documentation script and revise costing description so that user can be directed to description of unit model cost methods (#1064 )
- [ ] Documentation for the workflow for zero order modeling flowsheets - how to, after September 2022

**Nice to have**
- [ ] Disclaimer single warning when loading data from YAML for zero order model data @bknueven 
- [ ] Add process subtypes to docs. Some models have them, while others don't. Further, certain subtypes may have additional variables and constraints that should also be incorporated (e.g., ion_exchange) - after September @adam-a-a 
- [ ] Refine constraint docstrings that are either missing in unit models or could use improvement - @kurbansitterley 
- [ ] Noticed the electricity methods page could use a little more explanation- this is separate from the automated unit model documentation but worth noting nevertheless, as most unit model docs provide a link to the electricity helper methods doc. Reevaluate all original documentation on zero order modeling framework.
- [ ] Should document valid ranges for unit costing methods, prioritizing units with WaterTAP-adapted cost formulations. For example, add column(s) for valid ranges relevant to costing methods for document automation script. Could also apply to technical performance and valid param/var ranges, but this would be going above and beyond allocated time/resources. - After September
- [ ] add images for each unit, and/or some introductory sentence on what each unit can typically be applied for.
- [ ] Remove no index directive for class documentation and change to what we are currently doing with documentation.
- [ ] DELETE the "Smp" unit
- [ ] as @andrewlee94 mentioned, there are some tools that can display Pyomo expressions in LaTeX (https://github.com/IDAES/idaes-pse/blob/main/idaes/core/util/expr_doc.py); could include constraints in symbolic form in documentation; J Eslick designed those tools so follow up with him.

**DONE**
- [x] Disclaimer - any data for zero order models are for specific conditions, here are default values that could be useful but users need to get values that are for their application. Short description of what zero order model is and the warning. @adam-a-a @wigging PR #921 #977 
- [x] Add a check or test to ensure that all zero order models have a zero order doc; this should account for any zero order models that get added without being entered in the excel file from which models are read in for automating document creation (@lbianchi-lbl will be looking into this) after September @adam-a-a @MarcusHolly 
PR #799  and PR #877 
- [x] Update electricity and costing methods for relevant units.
- [x] Make exception for GasSpargedMembraneZO
- [x] Create .rst file for FeedZO @agarciadiego with @adam-a-a 
- [x] _Refine any funky rendering that took place via automating doc generation._ **(WASN'T AN ISSUE)**

### PR DESCRIPTION
**Pull request by adam-a-a**
## Summary/Motivation:
Revise zero order model documentation.

**Part 1/n: update documentation relevant to AMO models and add additional vars/constraints for all zero-order models**

## Changes proposed in this PR:
- update excel table to reflect latest AMO unit mods
- run automation script to update associated docs
- extend display of additional variables and constraints that were overlooked for all zero order models
**EDIT:**
- improve formatting by bolding model type, electricity functions, and cost methods; wrap var units in math directive and convert '**' to '^' to display superscripts.
-  replace "None" units with "dimensionless"

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: docs/technical_reference/unit_models/zero_order_unit_models/gas_sparged_membrane_zo.rst [hunk 1]
```diff
@@ -12,8 +13,25 @@ See documentation for :ref:`Helper Methods for Electricity Demand<electricity_me
...
-See documentation for the :ref:`zero-order costing package<zero_order_costing>`.
+This unit does not include costing.
+
+Additional Variables
+--------------------
+
+.. csv-table::
+   :header: "Description", "Variable Name", "Units"
+
+   "Mass flow of gas extracted per mass flow of influent", "gas_mass_influent_ratio", "None"
```
Comment (andrewlee94) [inline docs/technical_reference/unit_models/zero_order_unit_models/gas_sparged_membrane_zo.rst]: Minor comment - would this be better rendered as `dimensionless` than `None`? Whilst they probably mean the same thing to most users, from a technical standpoint they are slightly different (explicitly dimensionless versus no units assigned).

File: docs/technical_reference/unit_models/zero_order_unit_models/landfill_zo.rst [hunk 2]
```diff
@@ -16,6 +16,23 @@ Costing Method
...
+   "capacity basis for capital cost", "capacity_basis", "kg/hr"
+   "total mass flow rate", "total_mass", "kg/hr"
+
+Additional Constraints
+----------------------
+
+.. csv-table::
+   :header: "Description", "Constraint Name"
+
+   "Total mass constraint", "total_mass_constraint"
```
Comment (andrewlee94) [inline docs/technical_reference/unit_models/zero_order_unit_models/landfill_zo.rst:34]: Another passing comment - I believe there are tools in IDAES that can render Pyomo expressions as LaTeX which might allow us to include the actual expression in the docs if we want to.

File: docs/technical_reference/unit_models/zero_order_unit_models/mabr_zo.rst [hunk 3]
```diff
@@ -16,6 +16,26 @@ Costing Method
...
 See documentation for the :ref:`zero-order costing package<zero_order_costing>`.
 
+Additional Variables
+--------------------
+
+.. csv-table::
+   :header: "Description", "Variable Name", "Units"
+
+   "Nitrogen removal rate per day", "nitrogen_removal_rate", "g/d/m**2"
+   "Sizing variable for effective reactor area", "reactor_area", "m**2"
```
Comment (andrewlee94) [inline docs/technical_reference/unit_models/zero_order_unit_models/mabr_zo.rst]: This might just be GitHub, but this appeared in bold as I was reviewing it.

File: docs/technical_reference/unit_models/zero_order_unit_models/mbr_zo.rst [hunk 4]
```diff
@@ -16,6 +16,25 @@ Costing Method
...
 See documentation for the :ref:`zero-order costing package<zero_order_costing>`.
 
+Additional Variables
+--------------------
+
+.. csv-table::
+   :header: "Description", "Variable Name", "Units"
+
+   "Constant 1 in electricity intensity equation", "elec_coeff_1", "kWh/m**3"
+   "Constant 2 in electricity intensity equation", "elec_coeff_2", "None"
```
Comment (andrewlee94) [inline docs/technical_reference/unit_models/zero_order_unit_models/mbr_zo.rst]: Bold again.

**Review by andrewlee94 — COMMENTED** [review]
A few minor comments. The most significant one is that some of the lines were rendered in bold font in the GitHub review, but GitHub is also know to have the occasional issue with rendering rst files.

File: docs/technical_reference/unit_models/zero_order_unit_models/landfill_zo.rst (hunk 2 above)
Comment (adam-a-a) [reply docs/technical_reference/unit_models/zero_order_unit_models/landfill_zo.rst:34]: Can you point me to where those tools are?

Comment (andrewlee94) [reply docs/technical_reference/unit_models/zero_order_unit_models/landfill_zo.rst:34]: I think they might be here: https://github.com/IDAES/idaes-pse/blob/main/idaes/core/util/expr_doc.py

However, the best person to ask is @eslickj as these are some of his useful-but-no-so-well-documented tools.

File: docs/technical_reference/unit_models/zero_order_unit_models/anaerobic_mbr_mec_zo.rst [hunk 5]
```diff
@@ -8,12 +8,12 @@ See documentation for :ref:`reactive single-inlet, double-outlet Helper Methods<
 
 Electricity Consumption
 -----------------------
-Electricity consumption is calculated using the constant_intensity helper function.
+Electricity consumption is calculated using the pump_electricity helper function.
```
Comment (aladshaw3) [inline docs/technical_reference/unit_models/zero_order_unit_models/anaerobic_mbr_mec_zo.rst]:
```suggestion
Electricity consumption is calculated using the **pump_electricity** helper function.
```

It may be a good idea to **bold** the names of methods/functions so that in the rendered documentation they stand out more.

File: docs/technical_reference/unit_models/zero_order_unit_models/anaerobic_mbr_mec_zo.rst [hunk 6]
```diff
@@ -8,12 +8,12 @@ See documentation for :ref:`reactive single-inlet, double-outlet Helper Methods<
...
 See documentation for :ref:`Helper Methods for Electricity Demand<electricity_methods>`.
 
 Costing Method
 --------------
-Costing is calculated using the False method in the zero-order costing package.
+Costing is calculated using the cost_anaerobic_mbr_mec method in the zero-order costing package.
```
Comment (aladshaw3) [inline docs/technical_reference/unit_models/zero_order_unit_models/anaerobic_mbr_mec_zo.rst]:
```suggestion
Costing is calculated using the **cost_anaerobic_mbr_mec** method in the zero-order costing package.
```

File: docs/technical_reference/unit_models/zero_order_unit_models/cofermentation_zo.rst [hunk 7]
```diff
@@ -3,17 +3,17 @@ Cofermentation   (ZO)
...
 ----------
-This unit model is formulated as a double-input, single-output model form.
-See documentation for :ref:`double-input, single-output Helper Methods<diso_methods>`.
+This unit model is formulated as a reactive single-inlet, double-outlet model form.
+See documentation for :ref:`reactive single-inlet, double-outlet Helper Methods<sidor_methods>`.
 
 Electricity Consumption
 -----------------------
-Electricity consumption is calculated using the constant_intensity helper function.
+Electricity consumption is calculated using the pump_electricity helper function.
```
Comment (aladshaw3) [inline docs/technical_reference/unit_models/zero_order_unit_models/cofermentation_zo.rst]:
```suggestion
Electricity consumption is calculated using the **pump_electricity** helper function.
```

File: docs/technical_reference/unit_models/zero_order_unit_models/cofermentation_zo.rst [hunk 8]
```diff
@@ -3,17 +3,17 @@ Cofermentation   (ZO)
...
 See documentation for :ref:`Helper Methods for Electricity Demand<electricity_methods>`.
 
 Costing Method
 --------------
-Costing is calculated using the False method in the zero-order costing package.
+Costing is calculated using the cost_cofermentation method in the zero-order costing package.
```
Comment (aladshaw3) [inline docs/technical_reference/unit_models/zero_order_unit_models/cofermentation_zo.rst]:
```suggestion
Costing is calculated using the **cost_cofermentation** method in the zero-order costing package.
```

File: docs/technical_reference/unit_models/zero_order_unit_models/constructed_wetlands_zo.rst [hunk 9]
```diff
@@ -8,12 +8,11 @@ See documentation for :ref:`single-input, single-output Helper Methods<siso_meth
...
 Electricity Consumption
 -----------------------
-Electricity consumption is calculated using the -- helper function.
-See documentation for :ref:`Helper Methods for Electricity Demand<electricity_methods>`.
+This unit does not include energy consumption.
 
 Costing Method
 --------------
-Costing is calculated using the False method in the zero-order costing package.
+Costing is calculated using the cost_constructed_wetlands method in the zero-order costing package.
```
Comment (aladshaw3) [inline docs/technical_reference/unit_models/zero_order_unit_models/constructed_wetlands_zo.rst]:
```suggestion
Costing is calculated using the **cost_constructed_wetlands** method in the zero-order costing package.
```

File: docs/technical_reference/unit_models/zero_order_unit_models/electrochemical_nutrient_removal_zo.rst [hunk 10]
```diff
@@ -8,14 +8,33 @@ See documentation for :ref:`reactive single-inlet, double-outlet Helper Methods<
 
 Electricity Consumption
 -----------------------
-Electricity consumption is calculated using the f(x) helper function.
-See documentation for :ref:`Helper Methods for Electricity Demand<electricity_methods>`.
+The constraint used to calculate energy consumption is described in the Additional Constraints section below. More details can be found in the unit model class.
 
 Costing Method
 --------------
 Costing is calculated using the cost_electrochemical_nutrient_removal method in the zero-order costing package.
```
Comment (aladshaw3) [inline docs/technical_reference/unit_models/zero_order_unit_models/electrochemical_nutrient_removal_zo.rst]:
```suggestion
Costing is calculated using the **cost_electrochemical_nutrient_removal** method in the zero-order costing package.
```

File: docs/technical_reference/unit_models/zero_order_unit_models/landfill_zo.rst [hunk 11]
```diff
@@ -16,6 +16,23 @@ Costing Method
 Costing is calculated using the cost_landfill method in the zero-order costing package.
```
Comment (aladshaw3) [inline docs/technical_reference/unit_models/zero_order_unit_models/landfill_zo.rst]:
```suggestion
Costing is calculated using the **cost_landfill** method in the zero-order costing package.
```

File: docs/technical_reference/unit_models/zero_order_unit_models/mabr_zo.rst [hunk 12]
```diff
@@ -16,6 +16,26 @@ Costing Method
 Costing is calculated using the cost_mabr method in the zero-order costing package.
```
Comment (aladshaw3) [inline docs/technical_reference/unit_models/zero_order_unit_models/mabr_zo.rst]:
```suggestion
Costing is calculated using the **cost_mabr** method in the zero-order costing package.
```

File: docs/technical_reference/unit_models/zero_order_unit_models/mabr_zo.rst (hunk 3 above)
Comment (aladshaw3) [reply docs/technical_reference/unit_models/zero_order_unit_models/mabr_zo.rst]: I believe this is a GitHub rendering issue caused by the '**' being used. 

It may be a good idea to convert this '**' into a '^', or to use an actual superscript `(\ :sup:`2`).`

File: docs/technical_reference/unit_models/zero_order_unit_models/gas_sparged_membrane_zo.rst (hunk 1 above)
Comment (adam-a-a) [reply docs/technical_reference/unit_models/zero_order_unit_models/gas_sparged_membrane_zo.rst]: The thing here is I am automatically pulling out the units from each variable for every model with  `model_var._units`, so this is what comes out for dimensionless units. I could add a string replacement to swap "None" with "dimensionless".

File: docs/technical_reference/unit_models/zero_order_unit_models/mabr_zo.rst (hunk 3 above)
Comment (adam-a-a) [reply docs/technical_reference/unit_models/zero_order_unit_models/mabr_zo.rst]: I originally thought this was an issue too when looking at this through Github. However, when I checked via read the docs, the rendering was fine.

File: docs/technical_reference/unit_models/zero_order_unit_models/mbr_zo.rst (hunk 4 above)
Comment (adam-a-a) [reply docs/technical_reference/unit_models/zero_order_unit_models/mbr_zo.rst]: Just an issue in GH

File: docs/technical_reference/unit_models/zero_order_unit_models/mabr_zo.rst (hunk 3 above)
Comment (aladshaw3) [inline docs/technical_reference/unit_models/zero_order_unit_models/mabr_zo.rst]:
```suggestion
   "Nitrogen removal rate per day", "nitrogen_removal_rate", "g/d/m\ :sup:`2`"
   "Sizing variable for effective reactor area", "reactor_area", "m\ :sup:`2`"
```

File: docs/technical_reference/unit_models/zero_order_unit_models/mabr_zo.rst [hunk 13]
```diff
@@ -16,6 +16,26 @@ Costing Method
...
+Additional Variables
+--------------------
+
+.. csv-table::
+   :header: "Description", "Variable Name", "Units"
+
+   "Nitrogen removal rate per day", "nitrogen_removal_rate", "g/d/m**2"
+   "Sizing variable for effective reactor area", "reactor_area", "m**2"
+   "Air flow rate per area", "air_flow_rate", "m/hr"
+   "Volumetric air flow rate", "air_flow_vol", "m**3/hr"
```
Comment (aladshaw3) [inline docs/technical_reference/unit_models/zero_order_unit_models/mabr_zo.rst]:
```suggestion
   "Volumetric air flow rate", "air_flow_vol", "m\ :sup:`3`\ /hr"
```

File: docs/technical_reference/unit_models/zero_order_unit_models/mbr_zo.rst [hunk 14]
```diff
@@ -16,6 +16,25 @@ Costing Method
 Costing is calculated using the cost_power_law_flow method in the zero-order costing package.
 See documentation for the :ref:`zero-order costing package<zero_order_costing>`.
 
+Additional Variables
+--------------------
+
+.. csv-table::
+   :header: "Description", "Variable Name", "Units"
+
+   "Constant 1 in electricity intensity equation", "elec_coeff_1", "kWh/m**3"
```
Comment (aladshaw3) [inline docs/technical_reference/unit_models/zero_order_unit_models/mbr_zo.rst]:
```suggestion
   "Constant 1 in electricity intensity equation", "elec_coeff_1", "kWh/m\ :sup:`3`"
```

File: docs/technical_reference/unit_models/zero_order_unit_models/mbr_zo.rst [hunk 15]
```diff
@@ -16,6 +16,25 @@ Costing Method
...
+   "Constant 2 in electricity intensity equation", "elec_coeff_2", "None"
+   "Specific energy consumption with respect to feed flowrate", "electricity_intensity", "kWh/m**3"
```
Comment (aladshaw3) [inline docs/technical_reference/unit_models/zero_order_unit_models/mbr_zo.rst]:
```suggestion
   "Specific energy consumption with respect to feed flowrate", "electricity_intensity", "kWh/m\ :sup:`3`"
```

File: docs/technical_reference/unit_models/zero_order_unit_models/gas_sparged_membrane_zo.rst (hunk 1 above)
Comment (adam-a-a) [reply docs/technical_reference/unit_models/zero_order_unit_models/gas_sparged_membrane_zo.rst]: FYI the line I am referring to (L140) is here:
https://github.com/watertap-org/watertap/blob/b901e76ffdd2badeeabb7055439d3418d2f01337/docs/technical_reference/unit_models/zero_order_unit_models/automate_rst_file_wt3.py#L140

File: docs/technical_reference/unit_models/zero_order_unit_models/metab_zo.rst [hunk 16]
```diff
@@ -8,14 +8,37 @@ See documentation for :ref:`reactive single-inlet, double-outlet Helper Methods<
...
 Costing is calculated using the cost_metab method in the zero-order costing package.
 See documentation for the :ref:`zero-order costing package<zero_order_costing>`.
 
+Additional Variables
+--------------------
+
+.. csv-table::
+   :header: "Description", "Variable Name", "Units"
+
+   "Reactor volume", "volume", "m**3"
```
Comment (aladshaw3) [inline docs/technical_reference/unit_models/zero_order_unit_models/metab_zo.rst]:
```suggestion
   "Reactor volume", "volume", "m\ :sup:`3`"
```

File: docs/technical_reference/unit_models/zero_order_unit_models/metab_zo.rst [hunk 17]
```diff
@@ -8,14 +8,37 @@ See documentation for :ref:`reactive single-inlet, double-outlet Helper Methods<
...
+   "Hydraulic residence time", "hydraulic_retention_time", "hr"
+   "Electricity demand of unit", "electricity", "kW"
+   "Thermal demand of unit", "heat", "kW"
+   "Electricity intensity of mixer with respect to reactor volume", "energy_electric_mixer_vol", "kW/m**3"
```
Comment (aladshaw3) [inline docs/technical_reference/unit_models/zero_order_unit_models/metab_zo.rst]:
```suggestion
   "Electricity intensity of mixer with respect to reactor volume", "energy_electric_mixer_vol", "kW/m\ :sup:`3`"
```

File: docs/technical_reference/unit_models/zero_order_unit_models/metab_zo.rst [hunk 18]
```diff
@@ -8,14 +8,37 @@ See documentation for :ref:`reactive single-inlet, double-outlet Helper Methods<
...
+   "Electricity intensity of vacuum pump with respect to product gas flow", "energy_electric_vacuum_flow_vol_byproduct", "hr*kW/kg"
+   "Thermal energy intensity of reactor with respect to inlet volumetric flowrate", "energy_thermal_flow_vol_inlet", "kJ/m**3"
```
Comment (aladshaw3) [inline docs/technical_reference/unit_models/zero_order_unit_models/metab_zo.rst]:
```suggestion
   "Thermal energy intensity of reactor with respect to inlet volumetric flowrate", "energy_thermal_flow_vol_inlet", "kJ/m\ :sup:`3`"
```

File: docs/technical_reference/unit_models/zero_order_unit_models/photothermal_membrane_zo.rst [hunk 19]
```diff
@@ -8,14 +8,30 @@ See documentation for :ref:`single-input, double-output Helper Methods<sido_meth
 
 Electricity Consumption
 -----------------------
-Electricity consumption is calculated using the f(x) helper function.
-See documentation for :ref:`Helper Methods for Electricity Demand<electricity_methods>`.
+The constraint used to calculate energy consumption is described in the Additional Constraints section below. More details can be found in the unit model class.
 
 Costing Method
 --------------
 Costing is calculated using the cost_photothermal_membrane method in the zero-order costing package.
```
Comment (aladshaw3) [inline docs/technical_reference/unit_models/zero_order_unit_models/photothermal_membrane_zo.rst]:
```suggestion
Costing is calculated using the **cost_photothermal_membrane** method in the zero-order costing package.
```

File: docs/technical_reference/unit_models/zero_order_unit_models/photothermal_membrane_zo.rst [hunk 20]
```diff
@@ -8,14 +8,30 @@ See documentation for :ref:`single-input, double-output Helper Methods<sido_meth
...
 See documentation for the :ref:`zero-order costing package<zero_order_costing>`.
 
+Additional Variables
+--------------------
+
+.. csv-table::
+   :header: "Description", "Variable Name", "Units"
+
+   "Water flux through membrane", "water_flux", "kg/hr/m**2"
```
Comment (aladshaw3) [inline docs/technical_reference/unit_models/zero_order_unit_models/photothermal_membrane_zo.rst]:
```suggestion
   "Water flux through membrane", "water_flux", "kg/hr/m\ :sup:`2`"
```

File: docs/technical_reference/unit_models/zero_order_unit_models/photothermal_membrane_zo.rst [hunk 21]
```diff
@@ -8,14 +8,30 @@ See documentation for :ref:`single-input, double-output Helper Methods<sido_meth
...
+   "Membrane area", "membrane_area", "m**2"
```
Comment (aladshaw3) [inline docs/technical_reference/unit_models/zero_order_unit_models/photothermal_membrane_zo.rst]:
```suggestion
   "Membrane area", "membrane_area", "m\ :sup:`2`"
```

File: docs/technical_reference/unit_models/zero_order_unit_models/vfa_recovery_zo.rst [hunk 22]
```diff
@@ -13,9 +13,26 @@ See documentation for :ref:`Helper Methods for Electricity Demand<electricity_me
 
 Costing Method
 --------------
-Costing is calculated using the False method in the zero-order costing package.
+Costing is calculated using the cost_vfa_recovery method in the zero-order costing package.
```
Comment (aladshaw3) [inline docs/technical_reference/unit_models/zero_order_unit_models/vfa_recovery_zo.rst]:
```suggestion
Costing is calculated using the **cost_vfa_recovery** method in the zero-order costing package.
```

**Review by aladshaw3 — COMMENTED** [review]
It looks good. I just have some minor suggestions on formatting to make it look nicer.

File: docs/technical_reference/unit_models/zero_order_unit_models/gas_sparged_membrane_zo.rst (hunk 1 above)
Comment (andrewlee94) [reply docs/technical_reference/unit_models/zero_order_unit_models/gas_sparged_membrane_zo.rst]: It might be that the issue is in the underlying variable then (i.e. it wasn't assigned units).

**Comment by aladshaw3** [comment]
@adam-a-a 

After a second look at this, I am not sure how difficult it may be to automate some of the changes I am suggesting may be (specifically the superscripting for the units). So, I think that part can be left as is. 

It shouldn't be too difficult, however, to make it so that the names of methods are bolded in the rst files. This might be a worthwhile change, but overall it looks fine.

File: docs/technical_reference/unit_models/zero_order_unit_models/gas_sparged_membrane_zo.rst (hunk 1 above)
Comment (adam-a-a) [reply docs/technical_reference/unit_models/zero_order_unit_models/gas_sparged_membrane_zo.rst]: Looking at this one example for `gas_mass_influent_ratio`, the units were assigned as dimensionless.

Comment (bknueven) [reply docs/technical_reference/unit_models/zero_order_unit_models/gas_sparged_membrane_zo.rst]: No -- dimensionless variables have units of `None`, even if explicitly specified.

We could just replace `"None"` with `"dimensionless"` in our string processing, however, e.g.:
```python
added_var_units.append(str(model_var._units).replace("'", "").replace("None", "dimensionless")) 
```
or better:
```python
var_units = model_var._units
if var_units is None:
    added_var_units.append("dimensionless")
else:
    added_var_units.append(str(var_units).replace("'", ""))
```

File: docs/technical_reference/unit_models/zero_order_unit_models/anaerobic_mbr_mec_zo.rst (hunk 5 above)
Comment (adam-a-a) [reply docs/technical_reference/unit_models/zero_order_unit_models/anaerobic_mbr_mec_zo.rst]: Good idea- I'll add that to the automate script and rerun

File: docs/technical_reference/unit_models/zero_order_unit_models/gas_sparged_membrane_zo.rst (hunk 1 above)
Comment (andrewlee94) [reply docs/technical_reference/unit_models/zero_order_unit_models/gas_sparged_membrane_zo.rst]: An alternative (which might be better):

```
from pyomo.environ import as_quantity

added_var_units.append(str(as_quantity(model_var).u))
```

File: docs/technical_reference/unit_models/zero_order_unit_models/anaerobic_mbr_mec_zo.rst (hunk 6 above)
Comment (adam-a-a) [reply docs/technical_reference/unit_models/zero_order_unit_models/anaerobic_mbr_mec_zo.rst]: Added this too in the automate script (also bolding the "model type")

File: docs/technical_reference/unit_models/zero_order_unit_models/landfill_zo.rst (hunk 2 above)
Comment (adam-a-a) [reply docs/technical_reference/unit_models/zero_order_unit_models/landfill_zo.rst:34]: I had a look and (very quickly) tried using `document_constraints` and I got back some error that I was unsure of how to deal with. I'd be happy to discuss with [@eslickj](https://github.com/eslickj) in the future, but I think we'd better keep this for a subsequent PR.

File: docs/technical_reference/unit_models/zero_order_unit_models/mabr_zo.rst (hunk 3 above)
Comment (adam-a-a) [reply docs/technical_reference/unit_models/zero_order_unit_models/mabr_zo.rst]: I think I'll pass on these superscript mods for now since I am modifying everything through a single automation script, and this would require more effort than I think would be worthwhile at this point. I say this mostly because the bold rendering that GH shows, does _not_ occur in read the docs, so it is technically not an issue. However, it would be a nice-to-have to include actual superscripts rather than the `**` operator.

File: docs/technical_reference/unit_models/zero_order_unit_models/mabr_zo.rst (hunk 13 above)
Comment (adam-a-a) [reply docs/technical_reference/unit_models/zero_order_unit_models/mabr_zo.rst]: same response as above

File: docs/technical_reference/unit_models/zero_order_unit_models/gas_sparged_membrane_zo.rst (hunk 1 above)
Comment (adam-a-a) [reply docs/technical_reference/unit_models/zero_order_unit_models/gas_sparged_membrane_zo.rst]: Oh, I just saw your comments on this now. I already added something to handle this, though my solution may be more crude than either of your proposed solutions:

https://github.com/watertap-org/watertap/blob/89d6bf629915742588a472e05c7e1100f2bbfe3a/docs/technical_reference/unit_models/zero_order_unit_models/automate_rst_file_wt3.py#L140-L142

**Comment by adam-a-a** [comment]
@aladshaw3 - I managed to automate replacement of the superscripts, which I thought would be too hard to do quickly, but found an easier way by wrapping in a math directive. Also bolded model types, electricity functions, and cost methods.

@andrewlee94 @bknueven - I replaced "None" with "dimensionless" for documenting units. As for displaying LaTeX expressions for constraints, I think that'll have to wait until next time. I noted it as a "nice-to-have" in issue #595 .

Thanks for the pointers!

**Review by andrewlee94 — APPROVED** [review]
I only skimmed through, but I didn't see anything that looked wrong.

**Review by aladshaw3 — APPROVED** [review]
LGTM
